dlk/python/dlk/templates/manual/consts/input.tpl.cpp

```diff
@@ -28,4 +28,12 @@ limitations under the License.
   {%- endfor %}
 };
 
+{% if node.transposed_data %}
```

Comment:
I think `{{ node.name }}_transposed[]` is not need in C++ level, in order to reduce of memory size of `.elf` and `.so`, recommend to use only `{{ node.name }}[]` with `RUN_ON_FPGA` flag condition.

i.e.
```
{% if node.transposed_data %}

#if defined RUN_ON_FPGA
{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.transposed_data -%}
  {{- d -}},
  {%- endfor %}
};
#else
{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.data.flatten() -%}
  {{- d -}},
  {%- endfor %}
};

#endif

{% else -%}

{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.data.flatten() -%}
  {{- d -}},
  {%- endfor %}
};

{% endif %}
```

How is this ?

Reply:
@lm-konda 
What I thinking as a problem is that there are 2 input data (`{{ node.name }}_transposed[]` and `{{ node.name }}[] ` ) in `input.cpp`, but because we only use one of these at compile time, useless data to increase memory.

in node name input.cpp
```
{{ node.name }}_transposed[] = {
0.2, 0.3. ...............................................
};

{{ node.name }}[] = {
0.4, 0.2. ...............................................
};
```

We should avoid to increase memory with useless data.
How is to add `use_transposed_data_on_fpga ` at `Constant` operation and below snippets ?

```
{% if node.use_transposed_data_on_fpga %}

#if defined RUN_ON_FPGA
{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.transposed_data -%}
  {{- d -}},
  {%- endfor %}
};
#else
{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.data.flatten() -%}
  {{- d -}},
  {%- endfor %}
};

#endif

{% else -%}

{{ node.dtype.cpptype() }} {{ node.name }}[] = {
  {% for d in node.data.flatten() -%}
  {{- d -}},
  {%- endfor %}
};

{% endif %}
```

Reply:
I think `node.data.flatten()` include not only kernel data to be transposed but other data.
It's a little difficult problem.